### PR TITLE
Add static build to c-ares package with added shim static pkg-config file

### PR DESCRIPTION
**Pull request by irakhlin**
1. Modified libcares PKGBUILD for 2 round build, creating a static and shared library
2. Added patch to create a libcares-static.pc shim file that defines only the needed CFLAGS, and adding libcares-static as a Private dependency of libcares so it is linked and -DCARES_STATICLIB gets added correctly

Signed-off-by: irakhlin <i.rakhlin@gmail.com>

### CODE REVIEW

**Comment by irakhlin** [comment]
Only the 64bit TEA build specifically fails for  what looks like completely unrelated issues to the build..
err:seh:raise_exception Unhandled exception code c0000005 flags 0 addr 0x7bc9ca2e
checking types of args and return type for recvfrom...       6 [main] sh 284 fork: child -1 - CreateProcessW failed for 'C:\msys64\usr\bin\sh.exe', errno 13
../c-ares-1.13.0/configure: fork: Permission denied
make: *** No targets specified and no makefile found.  Stop.

I dont really whats happening on that build but it seems to work fine under appveyor

**Comment by mati865** [comment]
Tea runs outdated MSYS2 on Linux host, it's quite unstable.

File: mingw-w64-c-ares/PKGBUILD [hunk 1]
```diff
@@ -12,35 +12,58 @@ license=("MIT, BSD")
...
+  
+  msg "Building shared library..."
+  (cd "${srcdir}"/build-${CARCH}-shared
   ../${_realname}-${pkgver}/configure \
     --prefix=${MINGW_PREFIX} \
     --build=${MINGW_CHOST} \
     --host=${MINGW_CHOST} \
     --enable-shared \
+    --disable-static
+  make -j"$(nproc)"
```
Comment (Alexpux) [inline mingw-w64-c-ares/PKGBUILD]: Dont specify jobs num

File: mingw-w64-c-ares/PKGBUILD [hunk 2]
```diff
@@ -12,35 +12,58 @@ license=("MIT, BSD")
...
+  msg "Building static lib..."
+  (cd "${srcdir}"/build-${CARCH}-static
+  ../${_realname}-${pkgver}/configure \
+    --prefix=${MINGW_PREFIX} \
+    --build=${MINGW_CHOST} \
+    --host=${MINGW_CHOST} \
+    --disable-shared \
     --enable-static
-  make
+  make -j"$(nproc)"
```
Comment (Alexpux) [inline mingw-w64-c-ares/PKGBUILD]: As previous

**Comment by Alexpux** [comment]
Increase ````pkgrel```` for package rebuild

**Pull request merged**